File: 20241101-1.c
/*
// 연속 부분 수열 합의 개수 (https://school.programmers.co.kr/learn/courses/30/lessons/131701?language=c)

#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int elements[], size_t elements_len) {

    int* numbers = (int*)calloc(1000001, sizeof(int));
    int sum = 0;
    for (int i = 0; i < elements_len; i++) {
        sum += elements[i];
    }
    numbers[sum]++;

    int* nums = (int*)malloc(sizeof(int) * elements_len);
    for (int i = 0; i < elements_len; i++) {
        nums[i] = sum;
    }

    for (int i = 0; i < elements_len - 1; i++) {
        int n = i;
        for (int j = 0; j < elements_len; j++) {
            nums[j] -= elements[n];
            n++;
            if (n == elements_len)
                n = 0;
        }
        for (int j = 0; j < elements_len; j++) {
            numbers[nums[j]]++;
        }
    }
    free(nums);
    
    int answer = 0;
    for (int i = 0; i < 1000001; i++) {
        if (numbers[i] > 0)
            answer++;
    }
    free(numbers);

    return answer;

}

void main()
{

    int elements[] = { 7, 9, 1, 1, 4 };
    size_t elements_len = 5;
    printf("%d", solution(elements, elements_len));

}
*/

File: _20241101_1.java
// 괄호 회전하기 (https://school.programmers.co.kr/learn/courses/30/lessons/76502?language=java)
package _20241101_1;
import java.util.*;

class Solution {
    public int solution(String s) {

        int answer = 0;
        Deque<Character> deque = new ArrayDeque<>();

        for (int i = 0; i < s.length(); i++) {
            int t_f = 0;
            for (int j = 0; j < s.length(); j++) {
                if (s.charAt(j) == '(' || s.charAt(j) == '[' || s.charAt(j) == '{')
                    deque.addLast(s.charAt(j));
                else {
                    if (deque.isEmpty()) {
                        t_f = 1;
                        break;
                    } else {
                        if (s.charAt(j) == ')' && deque.getLast() == '(')
                            deque.removeLast();
                        else if (s.charAt(j) == ']' && deque.getLast() == '[')
                            deque.removeLast();
                        else if (s.charAt(j) == '}' && deque.getLast() == '{')
                            deque.removeLast();
                        else {
                            t_f = 1;
                            break;
                        }
                    }
                }
            }
            if (t_f == 0 && deque.isEmpty())
                answer++;
            deque.clear();
            s = s.substring(1) + s.substring(0, 1);
        }

        return answer;

    }
}

public class _20241101_1 {
    public static void main(String[] args) {

        Solution s = new Solution();
        System.out.println(s.solution("[](){}"));
        System.out.println(s.solution("}]()[{"));
        System.out.println(s.solution("[)(]"));
        System.out.println(s.solution("}}}"));

    }
}